pkg/executioncluster/impl/random_cluster_selector.go

```diff
@@ -123,51 +109,77 @@ func (s RandomClusterSelector) GetAllValidTargets() []executioncluster.Execution
 	return v
 }
 
-func (s RandomClusterSelector) GetTarget(spec *executioncluster.ExecutionTargetSpec) (*executioncluster.ExecutionTarget, error) {
-	if spec == nil || (spec.TargetID == "" && spec.ExecutionID == nil) {
-		return nil, fmt.Errorf("invalid executionTargetSpec %v", spec)
+func (s RandomClusterSelector) GetTarget(ctx context.Context, spec *executioncluster.ExecutionTargetSpec) (*executioncluster.ExecutionTarget, error) {
+	if spec == nil {
+		return nil, fmt.Errorf("empty executionTargetSpec")
 	}
 	if spec.TargetID != "" {
 		if val, ok := s.executionTargetMap[spec.TargetID]; ok {
 			return &val, nil
 		}
 		return nil, fmt.Errorf("invalid cluster target %s", spec.TargetID)
 	}
-	if spec.ExecutionID != nil {
-		if weightedRandomList, ok := s.domainWeightedRandomMap[spec.ExecutionID.GetDomain()]; ok {
-			executionName := spec.ExecutionID.GetName()
-			if executionName != "" {
-				randSrc, err := getRandSource(executionName)
-				if err != nil {
-					return nil, err
-				}
-				result, err := weightedRandomList.GetWithSeed(randSrc)
-				if err != nil {
-					return nil, err
-				}
-				execTarget := result.(executioncluster.ExecutionTarget)
-				return &execTarget, nil
-			}
-			execTarget := weightedRandomList.Get().(executioncluster.ExecutionTarget)
-			return &execTarget, nil
+	resource, err := s.resourceManager.GetResource(ctx, managerInterfaces.ResourceRequest{
+		Project:      spec.Project,
+		Domain:       spec.Domain,
+		Workflow:     spec.Workflow,
+		LaunchPlan:   spec.LaunchPlan,
+		ResourceType: admin.MatchableResource_EXECUTION_CLUSTER_LABEL,
+	})
+	if err != nil {
+		if flyteAdminError, ok := err.(errors.FlyteAdminError); !ok || flyteAdminError.Code() != codes.NotFound {
+			return nil, err
 		}
+		return nil, err
```

Comment:
Is this intended? The logic here is, if a resource is not found, an error will be returned, but it doesn't seem to be the intention given there is a null check for `resource` on line 136. My understanding is, if there is no override, and no label is configured, all defined clusters will be used as fallback.

Reply:
@anandswaminathan @katrogan

Reply:
@honnix you seem to have a point, the `return nil, err` in both cases
1. When err = Not Found
2. all other errors seems fishy.

Let me Talk to @katrogan and @anandswaminathan